flask_dance/contrib/gitlab.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import unicode_literals
+
+from flask_dance.consumer import OAuth2ConsumerBlueprint
+from functools import partial
+from flask.globals import LocalProxy, _lookup_app_object
+try:
+    from flask import _app_ctx_stack as stack
+except ImportError:
+    from flask import _request_ctx_stack as stack
+
+
+__maintainer__ = "Justin Georgeson <jgeorgeson@lopht.net>"
+
+
+def make_gitlab_blueprint(
+        client_id=None, client_secret=None, scope=None, redirect_url=None,
+        redirect_to=None, login_url=None, authorized_url=None,
+        session_class=None, backend=None, hostname="gitlab.com"):
```

Comment:
GitLab's biggest value proposition over GitHub is that you can run it on your own server, so I expect that this `hostname` parameter will be widely used. What do you think about moving it to the front of the arguments list, so that people could call it as a positional argument? For example:

```python
gitlab_bp = make_gitlab_blueprint("my-gitlab-installation.com")
```

Reply:
> GitHub also offers an self-hosted option, https://enterprise.github.com/home.

Hmm, that's true. (Didn't see that comment while I was reviewing this PR.) And now that I think about it, if we move this `hostname` parameter to the front, then that creates a different call signature compared to the `make_github_blueprint` factory -- and while we _could_ change the call signature of that factory as well, that would be a backwards-incompatible change, which I'd like to avoid.

What do you think? I see pros and cons to either approach.

Reply:
Putting it at the front was my initial instinct but I thought uniformity of the signature compared to the other providers, especially with respect to GitHub provider, was reason enough not to. But I'm fine with moving it to the front if you still prefer.